### PR TITLE
ci: Use faucet in actor tests

### PR DESCRIPTION
**Pull request by mslipper**
To improve the concurrency of the Bedrock actor tests, I hooked up an ETH faucet. Now, the actors can use the faucet to fund themselves before each run.

### CODE REVIEW

File: packages/actor-tests/lib/faucet.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+export class Faucet {
+  private url: string
+
+  private provider: providers.Provider
+
+  constructor(url: string, provider: providers.Provider) {
+    this.url = url
+    this.provider = provider
+  }
```
Comment (roninjin10) [inline packages/actor-tests/lib/faucet.ts:12]: Don't change this but in case you didn't know, you can write this with way less boilerplate

```typescript
constructor(
  private url: string,
  private provider: providers.Provider,
) {}
```

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
I haven't seen these run, but they look good to me

File: packages/actor-tests/bedrock/deposit-eth.ts [hunk 2]
```diff
@@ -1,71 +1,74 @@
-import { utils, Wallet, providers, constants } from 'ethers'
+import { utils, Wallet, constants } from 'ethers'
 import {
   CrossChainMessenger,
   ETHBridgeAdapter,
   StandardBridgeAdapter,
 } from '@eth-optimism/sdk'
-import { predeploys } from '@eth-optimism/contracts-bedrock'
+import { predeploys } from '@eth-optimism/contracts-bedrock/dist'
```
Comment (protolambda) [inline packages/actor-tests/bedrock/deposit-eth.ts:7]: Does this not work without `dist`?

**Review by protolambda — COMMENTED** [review]
LGTM, but wondering about one imports `dist` change, is that correct?

File: packages/actor-tests/tsconfig.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
     "resolveJsonModule": true
   },
   "include": [
-    "./lib",
+    "./lib/**/*.ts",
```
Comment (roninjin10) [inline packages/actor-tests/tsconfig.json:7]: This is wierd why is this necessary?

Comment (mslipper) [reply packages/actor-tests/tsconfig.json:7]: I was seeing compilation warnings.

Comment (roninjin10) [reply packages/actor-tests/tsconfig.json:7]: Odd.   I got no guesses on how this change would change the behavior of typescript

File: packages/actor-tests/bedrock/deposit-eth.ts (hunk 2 above)
Comment (mslipper) [reply packages/actor-tests/bedrock/deposit-eth.ts:7]: Nope, it didn't work without dist :(.

Comment (mslipper) [reply packages/actor-tests/bedrock/deposit-eth.ts:7]: Maybe @tynes knows how to fix it? It was trying to find an index file that didn't exist.

Comment (tynes) [reply packages/actor-tests/bedrock/deposit-eth.ts:7]: Try updating to the latest release? https://github.com/ethereum-optimism/optimism/blob/6f66c71f4163dc28f142ef87285631f4b64d2f05/packages/contracts-bedrock/package.json#L3

Your `package.json` refers to an older release. Other packages don't need to import from dist like this when working with this package?

**Comment by tynes** [comment]
If you rebase then it will get rid of the yarn.lock diff

**Comment by mslipper** [comment]
Will rebase + upgrade the package.

File: packages/actor-tests/bedrock/deposit-eth.ts (hunk 2 above)
Comment (mslipper) [reply packages/actor-tests/bedrock/deposit-eth.ts]: Fixed!

**Review by protolambda — APPROVED** [review]
